File: apps/backend/src/domain/item/ItemRepository.ts
import type { Item } from './Item.js'

export interface ItemRepository {
  findAll(): Promise<Item[]>
  findById(id: number): Promise<Item | null>
  save(item: Item): Promise<Item>
  update(id: number, item: Item): Promise<Item>
  delete(id: number): Promise<void>
}

File: apps/webshop/backend/src/main/java/com/frerememoire/webshop/domain/item/port/ItemRepository.java
package com.frerememoire.webshop.domain.item.port;

import com.frerememoire.webshop.domain.item.Item;

import java.util.List;
import java.util.Optional;

public interface ItemRepository {

    List<Item> findAll();

    Optional<Item> findById(Long id);

    Item save(Item item);

    void deleteById(Long id);

    boolean existsById(Long id);
}
